lib/sejfguru_web/controllers/asset_controller.ex

```diff
@@ -5,14 +5,18 @@ defmodule SejfguruWeb.AssetController do
   def index(conn, params) do
     conn
     |> assign(:page, fetch_page(params))
-    |> assign(:assets, fetch_assets(fetch_page(params)))
+    |> assign(:assets, fetch_assets(fetch_page(params), params["search"]))
     |> render("index.html")
   end
 
   defp fetch_page(%{"page" => page}), do: String.to_integer(page)
   defp fetch_page(_params), do: 1
 
-  defp fetch_assets(page) do
-    Sejfguru.Assets.list_assets_with_users(page: page)
+  defp fetch_assets(page, query) do
+    if query do
```

Comment:
this could be done by pattern matching in function, which I think is more elixir style than using conditions ;)
```elixir
defp fetch_assets(page, %{} = query), do: Sejfguru.Assets.filter_assets_with_users(query: query, page: page)
defp fetch_assets(page, nil), do: Sejfguru.Assets.list_assets_with_users(page: page)
```